storage/kubernetes/client.go

```diff
@@ -176,8 +189,19 @@ func (cli *client) getResource(apiVersion, namespace, resource, name string, v i
 	return json.NewDecoder(resp.Body).Decode(v)
 }
 
+func (cli *client) getResource(apiVersion, namespace, resource, name string, v interface{}) error {
+	return cli.getURL(cli.urlFor(apiVersion, namespace, resource, name), v)
+}
+
+func (cli *client) listN(resource string, v interface{}, n int) error {
+	params := url.Values{}
+	params.Add("limit", fmt.Sprintf("%d", n))
+	u := cli.urlForWithParams(cli.apiVersion, cli.namespace, resource, "", params)
+	return cli.getURL(u, v)
+}
+
 func (cli *client) list(resource string, v interface{}) error {
-	return cli.get(resource, "", v)
+	return cli.listN(resource, v, -1)
```

Comment:
Personally, I wouldn't say I like having `limit=-1` in all list requests besides GC.

The one option to solve the problem is to avoid `urlForWithParams` for the plain list method.
```go
func (cli *client) list(resource string, v interface{}) error {
	u := cli.urlFor(cli.apiVersion, cli.namespace, resource, "")
	return cli.getURL(u, v)
}
```

Reply:
Thought about this a bit more.  I don't think this necessarily needs to change in any form - was just trying to minimize recreating large blocks of code, but I think the other adjustments cover this anyway.